apps/cnquery/cmd/plugin.go

```diff
@@ -109,10 +109,12 @@ func (c *cnqueryPlugin) RunQuery(conf *run.RunQueryConfig, runtime *providers.Ru
 
 	for i := range assets {
 		connectAsset := assets[i]
-		if err := runtime.DetectProvider(connectAsset); err != nil {
+		// FIXME: I assume we need to transfer settings like recording, etc. to the new runtime
+		connectAssetRuntime := providers.Coordinator.NewRuntime()
```

Comment:
This comment led us to believe this might be the correct solution: https://github.com/mondoohq/cnquery/blob/1c014c0d3c383c0344e340ba75a0863e3843e6b0/providers/runtime.go#L20

> // Runtimes are associated with one asset and carry all providers
// and open connections for that asset.